src/hotspot/share/cds/filemap.cpp

```diff
@@ -2428,8 +2436,15 @@ void FileMapInfo::unmap_region(int i) {
     if (size > 0 && r->mapped_from_file()) {
       log_info(cds)("Unmapping region #%d at base " INTPTR_FORMAT " (%s)", i, p2i(mapped_base),
                     shared_region_name[i]);
-      if (!os::unmap_memory(mapped_base, size)) {
-        fatal("os::unmap_memory failed");
+      if (r->in_reserved_space()) {
+        // This region was mapped inside a ReservedSpace. Its memory will be freed when the ReservedSpace
+        // is released. Zero it so that we don't accidentally read its content.
+        log_info(cds)("Region #%d (%s) is in a reserved space, it will be freed when the space is released", i, shared_region_name[i]);
+        r->set_mapped_base(nullptr);
```

Comment:
There's no need for this line as it will be set to nullptr below line line 2450.